src/main/java/edu/ucsb/cs156/rec/controllers/RecommendationRequestController.java

```diff
@@ -0,0 +1,92 @@
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "RecommendationRequest")
+@RequestMapping("/api/recommendationrequest")
+@RestController
+public class RecommendationRequestController extends ApiController {
+    @Autowired
+    RecommendationRequestRepository recommendationRequestRepository;
+
+    /**
+     * This method returns a list of all recommendationrequests.
+     * @return a list of all recommendationrequests
+     */
+    @Operation(summary = "List all recommendation request")
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @GetMapping("/all")
+    public Iterable<RecommendationRequest> allRecommendationRequests() {
+        Iterable<RecommendationRequest> recommendationRequests = recommendationRequestRepository.findAll();
+        return recommendationRequests;
+    }
+
+    /**
+     * This method returns a single recommendationrequests.
+     * @param id id of the recommendationrequests to get
+     * @return a single recommendationrequests
+     */
+    @Operation(summary = "Get a single recommendation request")
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @GetMapping("")
+    public RecommendationRequest getById(
+            @Parameter(name = "id") @RequestParam Long id) {
+            RecommendationRequest recommendationRequest = recommendationRequestRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException(RecommendationRequest.class, id));
+
+        return recommendationRequest;
+    }
+
+    /**
+     * This method creates a new recommendationrequests. Accessible only to users with the role "ROLE_ADMIN" until our roles can be configured.
+     * @param professorName professor name of request
+     * @param professorEmail professor email of request
+     * @param recommendationTypes recommendation types of request
+     * @param details details of request
+     * @param submissionDate submission date of request
+     * @return the save recommendationrequests (with it's id field set by the database)
+     */
+    @Operation(summary = "Create a new recommendationrequests")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @PostMapping("/post")
+    public RecommendationRequest postRecommendationRequests(
+            @Parameter(name = "professorName") @RequestParam String professorName,
+            @Parameter(name = "professorEmail") @RequestParam String professorEmail,
+            @Parameter(name = "recommendationTypes") @RequestParam String recommendationTypes,
+            @Parameter(name = "details") @RequestParam String details,
+            @Parameter(name = "submissionDate") @RequestParam LocalDateTime submissionDate)
```

Comment:
Submission date should be computed automatically, either by getting the current time, or better yet, using this technique:

https://ucsb-cs156.github.io/topics/spring_boot/spring_boot_database.html#timestamps-on-database-rows

However, we probably do want to give users the option of putting in a deadline when their recommendation is needed (e.g. a due date).